project/src/main/java/com/google/sps/data/VolunteeringOpportunity.java

```diff
@@ -0,0 +1,79 @@
+package com.google.sps.data;
+
+import java.util.Set;
+
+public final class VolunteeringOpportunity {
+  private String name;
+  private int numSpotsLeft;
+  private Set<String> requiredSkills;
+  private Set<String> volunteers;
+
+  public static class Builder {
+    // Required parameters
+    private String name;
+    private int numSpotsLeft;
+
+    // Optional parameters
+    private Set<String> requiredSkills;
+    private Set<String> volunteers;
+
+    public Builder(String name, int numSpotsLeft) {
+      this.name = name;
+      this.numSpotsLeft = numSpotsLeft;
+    }
+
+    public Builder requiredSkills(Set<String> requiredSkills) {
+      this.requiredSkills = requiredSkills;
+      return this;
+    }
+
```

Comment:
Would there ever be a time where the user would edit and just add 1 required skill? if so we should have a addSkill()? Maybe that would make sense because right now we overwrite the entire Set, but also would the event host be able to edit their form and add skills?

Reply:
Done - yes we discussed this today.